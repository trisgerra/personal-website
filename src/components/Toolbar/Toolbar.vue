<template>
  <v-toolbar app>
    <v-toolbar-title class="headline text-uppercase">
      <img class="logo" width="63px" :src="logoPath"/>
    </v-toolbar-title>
    <v-spacer></v-spacer>
  </v-toolbar>
</template>


<script>
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class Toolbar extends Vue {
  logoPath = '../../assets/logo.png';
}
</script>

<style lang="scss" scoped>

.logo {
  object-fit: scale-down;
  height: 100%;
  }

.headline {
  background-color: #151616;
  padding: 0px 10px;
  height: 100%;
}
</style>
