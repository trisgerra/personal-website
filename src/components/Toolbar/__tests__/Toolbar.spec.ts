import { Wrapper } from '@vue/test-utils';
import Toolbar from '../Toolbar.vue';
import createWrapper from '@/utils/CreateWrapper';

describe('Toolbar', () => {
  it('should render as expected', () => {
    const wrapper: Wrapper<Toolbar> = createWrapper(Toolbar);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('contains the toolbar title', () => {
    const wrapper: Wrapper<Toolbar> = createWrapper(Toolbar);
    expect(wrapper.findAll('.headline').length).toBe(1);
  });

  it('contains the image logo', () => {
    const wrapper: Wrapper<Toolbar> = createWrapper(Toolbar);
    expect(wrapper.findAll('img').length).toBe(1);
    expect(wrapper.find('img').attributes('src')).toContain('../assets/logo.png');
  });
});
