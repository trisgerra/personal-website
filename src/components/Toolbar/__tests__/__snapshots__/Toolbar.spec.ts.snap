// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should render as expected 1`] = `
<nav
  class="v-toolbar v-toolbar--fixed theme--light"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
>
  <div
    class="v-toolbar__content"
    style="height: 64px;"
  >
    <div
      class="v-toolbar__title headline text-uppercase"
    >
      <img
        class="logo"
        src="../../assets/logo.png"
        width="63px"
      />
    </div>
     
    <div
      class="spacer"
    />
  </div>
</nav>
`;
